modules/core/lib/Storage/SQLPermanentStorage.php

```diff
@@ -94,7 +94,7 @@ public function set($type, $key1, $key2, $value, $duration = null)
      * @param int|null $duration
      * @return array
      */
-    private function insert($type, $key1, $key2, $value, $duration = null)
+    private function insert(string $type, $key1, $key2, $value, int $duration = null): array
```

Comment:
I guess since all three database-columns are defined as `text` we could assume strings, but that may cause some int-to-string ImplicitCast issues.. I'll give it a go and see what happens